### DISCUSSION TITLE
What is the recommended component folder structure in next 13?

### DISCUSSION THREAD

**Discussion by alextseitlin** [vercel/next.js#46409] [Help]
Before version 13 I would have something like this:

```
src
--- page
--- features
------ layout
--------- page
--------- header
--------- footer
------ signup
--- ui
------ form
------ link
```

I would have the basic components in 'ui' and more complex in 'features'. But now with v13 and 'app' folder I can place some of the components inside the app folder? What is the correct workflow now?

**Comment by kfirfitousi** [comment]
There is no one true answer for this.
If you want a feature-based structure, you can do something like this:

```
src/
├─ app/
│  ├─ signup/
│  │  ├─ sign-up-form.tsx  <- feature component
│  │  ├─ use-signup.tsx    <- feature hook
│  │  ├─ layout.tsx        <- feature layout
│  │  ├─ page.tsx          <- feature page
│  ├─ layout.tsx
│  ├─ page.tsx
├─ ui/                     <- shared components
│  ├─ link.tsx
│  ├─ form.tsx
├─ hooks/                  <- shared hooks
│  ├─ use-thing.tsx
```

Only `page.tsx` files are used as routes, so you can co-locate components and hooks inside the `app` directory.

**Reply by heloineto** [reply]
Hi there. I'm accustomed to seeing react components file names as "PascalCase" (like Footer.tsx, SignUpForm.tsx). And hooks / functions in "camelCase" (like useSignUp.ts).

I understand that React is unopinionated and that there is no right or wrong.

I'm asking out of curiosity, to know if you believe that because Next.js names it special files (page.tsx, layout.tsx) using "kebab-case", the other files should follow this convention. Is there a preferred/popupar/recommended approach to this?

**Reply by kfirfitousi** [reply]
I like using kebab-case for all file names, as a way of differentiating them from symbols (components, functions, types).
This also keeps all paths lowercase which avoids issues with case-sensitivity across different operating systems.

**Comment by pedroapfilho** [comment]
Is there any difference between using components inside or outside the `app/` folder? Should we use the ones that have `use client` inside the `app/` and the others outside? This is kinda confusing
